registration/register_node_led.go

```diff
@@ -230,12 +311,29 @@ func AuthorizeNode(
 		return nil, fmt.Errorf("(%s) authorize node cannot be called on an existing node", op)
 	}
 
+	return authorizeNodeCommon(ctx, storage, reqInfo, opt...)
+}
+
+// Note: this is called via paths that run the common validateFetchRequest
+// function which contains common validation functions
+func authorizeNodeCommon(
+	ctx context.Context,
+	storage nodeenrollment.Storage,
+	reqInfo *types.FetchNodeCredentialsInfo,
+	opt ...nodeenrollment.Option,
+) (*types.NodeInformation, error) {
+	const op = "nodeenrollment.registration.authorizeNodeCommon"
 	keyId, err := nodeenrollment.KeyIdFromPkix(reqInfo.CertificatePublicKeyPkix)
```

Comment:
Check for nil storage/ req?

Reply:
See the note above -- this is either called after validation or from the validation function itself.